saplings/circuits/src/components/Content.js

```diff
@@ -0,0 +1,155 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useReducer } from 'react';
+import { useLocalNodeState } from '../state/localNode';
+import mockCircuits from '../mockData/mockCircuits';
+import mockProposals from '../mockData/mockProposals';
+import { processCircuits } from '../data/processCircuits';
+
+import './Content.scss';
+
+const circuitsReducer = (state, action) => {
+  switch (action.type) {
+    case 'sort': {
+      const sortedCircuits = action.sortCircuits(
+        state.filteredCircuits,
+        action.sort
+      );
+      return { ...state, filteredCircuits: sortedCircuits };
+    }
+    case 'filter': {
+      const filteredCircuits = action.filterCircuits(
+        state.circuits,
+        action.filter
+      );
+      return { ...state, filteredCircuits };
+    }
+    default:
+      throw new Error(`unhandled action type: ${action.type}`);
+  }
+};
+
+const filterCircuits = (circuits, filterBy) => {
+  if (filterBy.filterTerm.length === 0) {
+    return circuits;
+  }
+  const filteredCircuits = circuits.filter(circuit => {
+    if (circuit.id.toLowerCase().indexOf(filterBy.filterTerm) > -1) {
+      return true;
+    }
+    if (
+      circuit.managementType.toLowerCase().indexOf(filterBy.filterTerm) > -1
+    ) {
+      return true;
+    }
+    if (circuit.comments.toLowerCase().indexOf(filterBy.filterTerm) > -1) {
+      return true;
+    }
+    if (
+      circuit.members.filter(
+        member => member.toLowerCase().indexOf(filterBy.filterTerm) > -1
+      ).length > 0
+    ) {
+      return true;
+    }
+    if (
+      circuit.roster.filter(
+        service => service.service_type.indexOf(filterBy.filterTerm) > -1
+      ).length > 0
+    ) {
+      return true;
+    }
+    return false;
+  });
+
+  return filteredCircuits;
+};
+
+const Content = () => {
+  const circuits = processCircuits(mockCircuits.concat(mockProposals));
+
+  const [circuitState, circuitsDispatch] = useReducer(circuitsReducer, {
+    circuits,
+    filteredCircuits: circuits
+  });
+  const nodeID = useLocalNodeState();
+  const totalCircuits = circuitState.circuits.length;
+  const actionRequired = circuitState.circuits.filter(circuit =>
+    circuit.actionRequired(nodeID)
+  ).length;
+
+  return (
+    <div className="content">
+      <div className="midContent">
+        <div className="circuit-stats">
+          <div className="stat total-circuits">
+            <span className="stat-count circuits-count">{totalCircuits}</span>
+            Total circuits
+          </div>
+          <div className="stat action-required">
+            <span className="stat-count action-required-count">
+              {actionRequired}
```

Comment:
This flashes to 5 before reverting to 2 on each reload. Probably an issue with the state management.

Reply:
That happened because of the call to get the local node ID, without a valid node ID the code would evaluate that all proposals needed to be voted on. I updated the code to show 0 if there is not a valid node id. I'm not sure if this is the best solution